core/drivers/crypto/crypto_api/hash/hash.c

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   Crypto Hash interface implementation to enable HW driver.
+ */
+#include <assert.h>
+#include <crypto/crypto_impl.h>
+#include <drvcrypt.h>
+#include <drvcrypt_hash.h>
+#include <malloc.h>
+#include <trace.h>
+#include <utee_defines.h>
+#include <util.h>
+
+/*
+ * Format the HASH context to keep the reference to the
+ * operation driver
+ */
+struct crypto_hash {
```

Comment:
Is this wrapper layer really needed? Is anything missing with `struct crypto_hash_ctx` which is stopping `drivers/crypto/caam/hash/caam_hash.c` from implementing a `struct crypto_hash_ctx` instead? Am I missing something?

Reply:
For sure, now it's no more needed as we reworked everything and removed the original checks. I can remove it but for the next crypto driver, I'll have the same structure and to be compliant I would prefer to keep this method. Now if you think that it will add complexity and overhead I can rework it to remove the `struct drvcrypt_hash`.
Please let me know.

Reply:
What do you mean by the next crypto driver? Compliant with what, in which context?

If it's only here for historical reasons I'd rather see it removed since it adds another level of indirection and makes it harder to follow the code.

Reply:
I mean the implementation for the Symmetric, Asymmetric and key derivation using the HW driver. There is the same wrapping operation structure used to reformat the crypto_xxx parameters in a data structure.
Anyway, for the Hash I'll remove it and we will see after for the others.

Reply:
I added a new commit (last one in the list) to the PR implementing this change. I'll squashed files after review and acceptance